### PR TITLE
Feature lys prepaid compatible

### PR DESCRIPTION
**Pull request by liyasongAha**
Pull Request PreCheck List:
- [x] rebase dev branch (**passed**)
- [x] make format & make build  (**passed**)
- [x] union test or test in dev env(**passed**)

Detail in branch
- [x] add `before_instance_count`and`expect_instance_count` to **`task/describe_all`** api
- [x] add `instance_type`and`charge_type` to **`cluster/describe_all`** api
- [x] add `count` and `beforeCount` to `TaskInfo`
- [x] check cluster if exist before `expand/shrink`
- [x] prevent `PrePaid` charge type instance to do `shrink`

### CODE REVIEW

File: internal/model/cluster.go [hunk 1]
```diff
@@ -37,6 +38,29 @@ type Cluster struct {
...
+		return ""
+	}
+	return conf.ChargeType
+}
+
+func (c *Cluster) UnmarshalChargeConfig() (*types.ChargeConfig, error) {
+	if c == nil {
+		return nil, nil
+	}
+	chargeConfig := &types.ChargeConfig{}
```
Comment (liqunn) [inline internal/model/cluster.go:56]: 这块统一下吧 我看上面是`types.ChargeConfig{}` 在unmarshal的时候取地址

File: internal/model/task.go [hunk 2]
```diff
@@ -27,12 +27,24 @@ func (Task) TableName() string {
 	return "task"
 }
 
+type TaskInfo interface {
+	GetExpectAndBeforeInstanceCount() (beforeCount int, expectCount int)
```
[the remark this answers is not in the document]
Comment (liyasongAha) [reply internal/model/task.go:31]: After 的话如果任务执行成功了那还挺合适的,但是有可能失败,所以 after 命名就不恰当了
`GetExpectAndBeforeInstanceCount`  ->  `GetBeforeAndExpectInstanceCount` 如何

Comment (liqunn) [inline internal/model/task.go:31]: 方法的顺序是`GetExpectAndBefore` 返回值顺序是before expect，可以统一下

Comment (liyasongAha) [reply internal/model/task.go]: 名称顺序改了下

**Review by liqunn — APPROVED** [review]

File: internal/model/cluster.go [hunk 3]
```diff
@@ -3,6 +3,7 @@ package model
 import (
 	"context"
 	"fmt"
+	"github.com/galaxy-future/BridgX/internal/errs"
```
Comment (liqunn) [inline internal/model/cluster.go:6]: goimports

File: cmd/api/helper/cluster.go [hunk 4]
```diff
@@ -1,6 +1,7 @@
 package helper
 
 import (
+	"github.com/galaxy-future/BridgX/internal/constants"
```
Comment (liqunn) [inline cmd/api/helper/cluster.go:4]: goimports

**Review by galaxy-future-liubo — APPROVED** [review]

**Pull request merged**